electrumsv/wallet.py

```diff
@@ -853,22 +888,23 @@ def set_transaction_state(self, tx_hash: bytes, flags: TxFlags) -> bool:
         self._wallet.trigger_callback('transaction_state_change',
             self._wallet.get_storage_path(), self._id, tx_hash, existing_flags, updated_flags)
 
-    def process_key_usage(self, tx_hash: bytes, tx: Transaction) -> None:
+    def process_key_usage(self, tx_hash: bytes, tx: Transaction,
+                          relevant_txos: Optional[List[Tuple[int, XTxOutput]]]) -> None:
         with self.transaction_lock:
-            self._process_key_usage(tx_hash, tx)
+            self._process_key_usage(tx_hash, tx, relevant_txos)
```

Comment:
What if the third argument was the active keyinstance_id (or not) and _process_key_usage looked up the outputs if provided it.

Reply:
Not sure about that. 

In some cases you really do want it to iterate over every output and add the txos to utxo set for example. So I prefer it with the caller having the contextual awareness to know when to provide relevant txos or when to not (and let if process all txos).